cpp/test/matrix/gather.cu

```diff
@@ -173,39 +194,53 @@ class GatherTest : public ::testing::TestWithParam<GatherInputs<IdxT>> {
+  TEST_P(test_name, Result)                                                                       \
+  {                                                                                               \
+    ASSERT_TRUE(                                                                                  \
+      devArrMatch(d_out_exp.data(), d_out_act.data(), d_out_exp.size(), raft::Compare<float>())); \
+  }                                                                                               \
   INSTANTIATE_TEST_CASE_P(GatherTests, test_name, ::testing::ValuesIn(test_inputs))
 
-const std::vector<GatherInputs<int>> inputs_i32 =
-  raft::util::itertools::product<GatherInputs<int>>({25, 2000}, {6, 31, 129}, {11, 999}, {1234ULL});
+const std::vector<GatherInputs<int>> inputs_i32 = raft::util::itertools::product<GatherInputs<int>>(
+  {25, 2000}, {6, 31, 129}, {11, 999}, {2, 3, 6}, {1234ULL});
 const std::vector<GatherInputs<int64_t>> inputs_i64 =
   raft::util::itertools::product<GatherInputs<int64_t>>(
-    {25, 2000}, {6, 31, 129}, {11, 999}, {1234ULL});
+    {25, 2000}, {6, 31, 129}, {11, 999}, {2, 3, 6}, {1234ULL});
+const std::vector<GatherInputs<int>> inplace_inputs_i32 =
+  raft::util::itertools::product<GatherInputs<int>>(
+    {2000}, {6, 31, 129}, {11, 999}, {2, 3, 6}, {1234ULL});
+const std::vector<GatherInputs<int64_t>> inplace_inputs_i64 =
+  raft::util::itertools::product<GatherInputs<int64_t>>(
+    {2000}, {6, 31, 129}, {11, 999}, {2, 3, 6}, {1234ULL});
 
-GATHER_TEST((GatherTest<false, false, float, uint32_t, int>), GatherTestFU32I32, inputs_i32);
-GATHER_TEST((GatherTest<false, true, float, uint32_t, int>),
+GATHER_TEST((GatherTest<false, false, false, float, uint32_t, int>), GatherTestFU32I32, inputs_i32);
+GATHER_TEST((GatherTest<false, true, false, float, uint32_t, int>),
             GatherTransformTestFU32I32,
             inputs_i32);
-GATHER_TEST((GatherTest<true, false, float, uint32_t, int>), GatherIfTestFU32I32, inputs_i32);
-GATHER_TEST((GatherTest<true, true, float, uint32_t, int>),
+GATHER_TEST((GatherTest<true, false, false, float, uint32_t, int>),
+            GatherIfTestFU32I32,
+            inputs_i32);
+GATHER_TEST((GatherTest<true, true, false, float, uint32_t, int>),
             GatherIfTransformTestFU32I32,
             inputs_i32);
-GATHER_TEST((GatherTest<true, true, double, uint32_t, int>),
+GATHER_TEST((GatherTest<true, true, false, double, uint32_t, int>),
             GatherIfTransformTestDU32I32,
             inputs_i32);
-GATHER_TEST((GatherTest<true, true, float, uint32_t, int64_t>),
+GATHER_TEST((GatherTest<true, true, false, float, uint32_t, int64_t>),
             GatherIfTransformTestFU32I64,
             inputs_i64);
-GATHER_TEST((GatherTest<true, true, float, int64_t, int64_t>),
+GATHER_TEST((GatherTest<true, true, false, float, int64_t, int64_t>),
             GatherIfTransformTestFI64I64,
             inputs_i64);
-
+GATHER_TEST((GatherTest<false, false, true, float, uint32_t, int>),
```

Comment:
Any reason we're only testing this small subset of possible template combinations? Are we unable to test all of the above for inplace as well?